### ISSUE TITLE
filter_pmap + Dropout/Pooling breaks under distributed training

### ISSUE THREAD

**Issue by RE-N-Y** [patrick-kidger/equinox#180]
First of all, thank you for the amazing library! I've been writing generative/vision models (StyleGAN, ESRGAN ...) in Equinox, but I've encountered a couple of issues with `filter_pmap`. 

```python
model = # model initialisation
devices = jax.local_devices()
pmodel = jax.device_put_replicated(model, devices)

def step(model, data, ...):
    out = model(data)
    # compute losses, pmean gradients, update gradients and etc ...
    return model, ...

model, ... = eqx.filter_pmap(step)(pmodel, data, ...)
```
However, there are a few cases where the above setup breaks.
1. `model` includes dropout operation
2. `model` includes pooling operation 

I've included a repro [here](https://colab.research.google.com/drive/1lFiOyv6mKdvcP_nNhLnqK5BxZqlNym8I?usp=sharing). For both of these issues, I solve them by:
1. Marking problematic properties (e.g. dropout's `p`) with `static_field` inside my Equinox Module
2. Replace pooling operation with custom einop's `reduce` operation

In earlier versions of Equinox I had similar issues when my model involved non-JAX types. So, I maunally mark properties with `static-field`. I know it's not the recommended way to use Equinox. It would be nice if the documentation included a working example with model involving non-JAX type + filter_pmap for parallel traning.

**Comment by RE-N-Y** [comment]
Thank you! The setup that you suggested above seems to resolve the issue. I might as well as start all removing those `static_field` I put into my model definitions...
